sandbox/components/PageHeader.vue

```diff
@@ -38,21 +39,24 @@ const emit = defineEmits<{
 
 const query = ref('')
 
-watch(query, (searchQuery: string) => {
+const handleQueryUpdate = debounce((searchQuery: string) => {
   emit('search', searchQuery)
 
   if (searchQuery) {
-    if (searchQuery !== route.query.search) {
-      router.push({ name: 'home', query: { search: searchQuery } })
-    }
+    router.push({ name: 'home', query: { q: searchQuery } })
   } else {
     router.push({ name: 'home' })
   }
+}, 1000)
```

Comment:
suggestion: we could decrease this to maybe `300` or `500` ?

Reply:
No objections to that, let's do 500